third_party/haskell/BUILD.vault

```diff
@@ -0,0 +1,34 @@
+package(default_visibility = ["//visibility:public"])
+
+load("@io_tweag_rules_haskell//haskell:haskell.bzl", "haskell_library")
+load("@ai_formation_hazel//:hazel.bzl", "hazel_library")
+
+haskell_library(
+  name = "vault",
+  srcs = [
```

Comment:
I think the reason `vault` doesn't work is that Hazel symlinks all the source files in a separate source directory, different from `srcs`.  Maybe fixing tweag/rules_haskell#168 (the `src_script_prefix` part of it) would let us avoid the symlinking and enable this to work.

Reply:
I'm not sure what you mean by "`vault` doesn't work". It builds fine on CI and locally. The CI failure is due to a file in `wai-app-static` with Unicode characters.

Reply:
I mean "why the rules that Hazel generates automatically for vault don't work" (missing the IO.hs import).

Reply:
Note that I updated the PR to use `vault` straight from Hackage instead of manually-patched one. Maybe this version will build automatically? Will need to try.